core/storage/chain/chain_data_store.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_STORAGE_CHAIN_CHAIN_DATA_STORE_HPP
+#define CPP_FILECOIN_CORE_STORAGE_CHAIN_CHAIN_DATA_STORE_HPP
+
+#include "storage/chain/datastore_key.hpp"
+
+namespace fc::storage::blockchain {
+
+  class ChainDataStore {
+   public:
+    virtual ~ChainDataStore() = default;
+    /** @brief get value by key */
+    virtual outcome::result<std::string> get(const DatastoreKey &key) const = 0;
+    /** @brief set key -> value */
+    virtual outcome::result<void> set(const DatastoreKey &key,
+                                      std::string_view value) = 0;
```

Comment:
will there be any string validation, it's quite strange if it possible to put any string value to a chain data store

Reply:
I don't think it is strange. It looks normal for me. What kind of validation you suggest?